cli/context/kubernetes/load.go

```diff
@@ -61,6 +74,25 @@ func (c *Endpoint) KubernetesConfig() clientcmd.ClientConfig {
 	return clientcmd.NewDefaultClientConfig(*cfg, &clientcmd.ConfigOverrides{})
 }
 
+// ResolveDefault returns endpoint metadata for the default Kubernetes
+// endpoint, which is derived from the env-based kubeconfig.
+func (c *EndpointMeta) ResolveDefault() (interface{}, *store.EndpointTLSData) {
+	kubeconfig := os.Getenv("KUBECONFIG")
+	if kubeconfig == "" {
+		kubeconfig = filepath.Join(homedir.Get(), ".kube/config")
+	}
+	kubeEP, err := FromKubeConfig(kubeconfig, "", "")
+	if err != nil {
+		return nil, nil
```

Comment:
@simonferquel this is refactoring the behaviour of the original code which was to ignore failures here (commonly "file not found") was this a deliberate choice in the original?

I originally had this return the `err` but then things break if you haven't got a `$HOME/.kube` (which happens in the `test` target somewhere). I considered whitelisting some errors ("not found", maybe one or two others, like "not a director" perhaps) but for now settled on just replicating the original behaviour. WDYT?

Reply:
Yes, we should not break the docker CLI if we have no kubeconfig file, or if it is damanged: in those cases, we should just fail running Kubernetes related commands.